examples/seq2seq/seq2seq_trainer.py

```diff
@@ -114,23 +123,31 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.sampler.Sampler]:
                 else DistributedSampler(self.train_dataset)
             )
 
-    def compute_loss(self, model, inputs):
-        labels = inputs.pop("labels")
-        outputs = model(**inputs, use_cache=False)
-        logits = outputs[0]
-        return self._compute_loss(logits, labels)
-
-    def _compute_loss(self, logits, labels):
+    def _compute_loss(self, model, inputs):
         if self.args.label_smoothing == 0:
-            # Same behavior as modeling_bart.py
-            loss_fct = torch.nn.CrossEntropyLoss(ignore_index=self.config.pad_token_id)
-            assert logits.shape[-1] == self.vocab_size
-            loss = loss_fct(logits.view(-1, logits.shape[-1]), labels.view(-1))
+            # set all ids to -100 to be ignored
+            if self.data_args.ignore_pad_token_for_loss:
+                assert self.config.pad_token_id >= 0, "Make sure that `config.pad_token_id` is correcly defined"
+                inputs["labels"][inputs["labels"] == self.config.pad_token_id] = -100
+
+            # compute usual loss via models
+            loss, logits = model(**inputs)[:2]
         else:
+            assert (
+                self.config.pad_token_id >= 0
+            ), "Label smoothing is currently only supported if `config.pad_token_id` is defined."
+            # compute label smoothed loss
+            inputs = copy.deepcopy(inputs)
+            labels = inputs.pop("labels")
+            logits = model(**inputs, use_cache=False)[0]
             lprobs = torch.nn.functional.log_softmax(logits, dim=-1)
-            loss, nll_loss = label_smoothed_nll_loss(
+            loss, _ = label_smoothed_nll_loss(
                 lprobs, labels, self.args.label_smoothing, ignore_index=self.config.pad_token_id
             )
+        return loss, logits
+
+    def compute_loss(self, model, inputs):
```

Comment:
Why two functions with same signature?

Reply:
same as before ;-) for better re-useability